### PR TITLE
diagnostic commit

### PR DESCRIPTION
**Pull request by kpie89**

### CODE REVIEW

File: app/models/licensee.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+class Licensee < ActiveRecord::Base
+end
```
Comment (GA-MEB) [inline app/models/licensee.rb:2]: Model : Check

File: db/migrate/20160219140815_create_licensees.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+      t.integer :height
+      t.integer :weight
+      t.date :born_on
+      t.string :eye_color
+      t.string :hair_color
+
+      t.timestamps null: false
+    end
+  end
+end
```
Comment (GA-MEB) [inline db/migrate/20160219140815_create_licensees.rb:16]: Migration : Check

File: db/schema.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    t.integer  "height"
+    t.integer  "weight"
+    t.date     "born_on"
+    t.string   "eye_color"
+    t.string   "hair_color"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+  end
+
+end
```
Comment (GA-MEB) [inline db/schema.rb:32]: Ran Migration : Check

File: diagnostic.rb [hunk 4]
```diff
@@ -2,15 +2,26 @@
...
 # question 2:
-
+Licensee.create surname: 'Bishop',
+                        given_name: 'Caren',
+                        gender: 'f',
+                        height: 63,
+                        weight: 132 ,
+                        born_on: '1943-09-26',
+                        eye_color: 'Brown',
+                        hair_color: 'Black'
```
Comment (GA-MEB) [inline diagnostic.rb:12]: Correct.

File: diagnostic.rb [hunk 5]
```diff
@@ -2,15 +2,26 @@
...
 # question 3:
-Licensee.transaction do
-  CSV.foreach 'data/licensees.csv', headers: true do |row|
+Licensee.transaction dosss
+  CSV.foreach(Rails.root + 'data/licensees.csv',
+      headers: true) do |row|
+        licensee = row.to_hash
+                      Licensee.create! licensee unless Licensee.exists? licensee
```
Comment (GA-MEB) [inline diagnostic.rb:18]: Correct.

File: diagnostic.rb [hunk 6]
```diff
@@ -2,15 +2,26 @@
...
   end
 end
 
 # question 4:
-
-# question 5:
+Licensee.where eye_color: 'Hazel', hair_color: 'Brown', hair_color: 'Black'
```
Comment (GA-MEB) [inline diagnostic.rb:23]: Not quite. `eye_color: 'Hazel', hair_color: 'Brown', hair_color: 'Black'` specifies a hash, and a hash's keys must be unique. You will only get either `hair_color: 'Brown'` _or_ `hair_color: 'Black'`. And having only one will not make the right query.

Comment (MattB-GA-Boston) [inline diagnostic.rb]: Using a query is probably the best option here.

File: diagnostic.rb [hunk 7]
```diff
@@ -2,15 +2,26 @@
...
+# question 5
 
 # question 6:
+Licensee.destroy given_name: 'Dylan', given_name: 'Teresita', surname: "Rich", surname: 'Myers'
```
Comment (GA-MEB) [inline diagnostic.rb:27]: See comment above re: hashes. You could delete one person by writing:
`Licensee.destroy given_name: 'Dylan', surname: 'Rich'`

**Comment by GA-MEB** [comment]
4 / 6 (-0.5 on Q4, no answer for Q5, -0.5 on Q6)

File: diagnostic.rb (hunk 6 above)
Comment (GA-MEB) [inline diagnostic.rb:23]: In this case, you could do the following:
`Licensee.where eye_color: 'Hazel', hair_color: ['Brown', 'Black']`
But if you wanted to do an OR on a combination of attributes, the best option is (currently) to use a query.

**Pull request closed without merging**